packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -91,6 +190,48 @@ export class DataProxyEngine extends Engine<DataProxyTxInfoPayload> {
   async start() {}
   async stop() {}
 
+  private propagateResponseExtensions(extensions: DataProxyExtensions): void {
+    const tracingConfig = getTracingConfig(this.config.previewFeatures || [])
+
+    if (extensions?.logs?.length) {
+      extensions.logs.forEach((log) => {
+        switch (log.level) {
+          case 'debug':
+          case 'error':
+          case 'trace':
+          case 'warn':
+          case 'info':
+            // TODO these are propagated into the response.errors key
+            break
+          case 'query': {
+            let dbQuery = typeof log.attributes.query === 'string' && log.attributes.query ? log.attributes.query : ''
```

Comment:
I believe we should now be able to drop the second clause in the `&&` this time (last time I probably wasn't really correct since I missed it could've been an empty string and there was a different value in the else branch):
```suggestion
            let dbQuery = typeof log.attributes.query === 'string ? log.attributes.query : ''
```

`log.attributes.query` is truthy if it's not null, undefined or an empty string
However, it is already checked that it is not null or undefined by checking `typeof log.attributes.query === 'string'`.

Thus the whole expression is equivalent to `typeof log.attributes.query === 'string' && log.attributes.query !== '' ? log.attributes.query : ''`, which is equivalent to `typeof log.attributes.query === 'string' ? log.attributes.query : ''`